### PR TITLE
fucking SEXTUPLES ALL TURF DECALS to test for if they affect maptick (like what i did before but more)

### PR DESCRIPTION
**Pull request by Kylerace**
## About The Pull Request
https://github.com/tgstation/tgstation/issues/53798 meant to test this, last time i did a test merge on this issue i got only a 0.07 maptick per person increase by doubling the number of turf decals over 65 rounds (0.882 for the 65 rounds before the test merge, 0.94 for the test merge, all on manuel). however im still unsure if what we got was within reasonable variance over 65 rounds since lummox says that overlays should have a null maptick cost and i cut off the test earlier than i wanted because of oom concerns, which doesnt apply now due to LAA on the servers. 

according to @LemonInTheDark this is our current maptick per person floor and ceiling. The variance is much lower now than when i last tried this thanks to 1556 and underlay lighting
Maptick_variance
if there really is a 5-10% difference on maptick caused by doubling turf decals, we should be able to be absolutely certain of it with 150-200 rounds of this test merged

## Why It's Good For The Game
science

### CODE REVIEW

**Comment by SmArtKar** [comment]
> Sextuples
nice

**Comment by Kylerace** [comment]
oh right this is what turf decals look like with this, the added ones have an alpha of 1 so theyre almost invisible
Screenshot_382

**Comment by interestingusernam3** [comment]
sex

**Comment by Time-Green** [comment]
C*ders literally adding lag

**Comment by LemonInTheDark** [comment]
Here's a slightly shorter time scale and more up to date version of that graph 
Maptick 3

The recent spike appears to have been a byond issue with map_cpu persisting through rounds by the by

**Comment by Kylerace** [comment]
> 
> The recent spike appears to have been a byond issue with map_cpu persisting through rounds by the by
cant wait for the inevitable: hey whats that super skinny spike on the right?

**Comment by maxymax13** [comment]
s*xtuples 😳

File: code/game/objects/effects/decals/decal.dm [hunk 1]
```diff
@@ -46,3 +46,9 @@
 	if(!istype(T)) //you know this will happen somehow
 		CRASH("Turf decal initialized in an object/nullspace")
 	T.AddElement(/datum/element/decal, icon, icon_state, dir, null, null, alpha, color, null, FALSE, null)
+
+	T.AddElement(/datum/element/decal, icon, icon_state, dir, null, null, 1, color, null, FALSE, null)
```
Comment (Iamgoofball) [inline code/game/objects/effects/decals/decal.dm:50]: no, wrong, bad, use a loop, its faster and cleaner

Comment (Mothblocks) [reply code/game/objects/effects/decals/decal.dm:50]: who cares its test code

**Comment by MrStonedOne** [comment]
this won't actually increase maptick thou

as far as byond and the client will see, the turf will have 1 extra overlay, not 6

File: code/game/objects/effects/decals/decal.dm (hunk 1 above)
Comment (MrStonedOne) [reply code/game/objects/effects/decals/decal.dm:50]: exactly. you can't extract any free lag from loops, they are already mined out.

**Comment by Iamgoofball** [comment]
ya it needs to be slightly different colors right?

**Comment by Kylerace** [comment]
> this won't actually increase maptick thou
> 
> as far as byond and the client will see, the turf will have 1 extra overlay, not 6

yeah i realized i was an idiot a bit ago since the 5 added elements are just the same one so it adds a single extra overlay. now all of them have different alpha values so they should all be additional overlays

File: code/game/objects/effects/decals/decal.dm (hunk 1 above)
Comment (Mothblocks) [reply code/game/objects/effects/decals/decal.dm:50]: loop unrolling is faster :(

**Comment by LemonInTheDark** [comment]
No

**Comment by Kylerace** [comment]
> 
> 
> This PR has been inactive for long enough to be automatically marked as stale. This means it is at risk of being auto closed in ~ 7 days, please address any outstanding review items and ensure your PR is finished, if these are all true and you are auto-staled anyway, you need to actively ask maintainers if your PR will be merged. Once you have done any of the previous actions then you should request a maintainer remove the stale label on your PR, to reset the stale timer. If you feel no maintainer will respond in that time, you may wish to close this PR youself, while you seek maintainer comment, as you will then be able to reopen the PR yourself

https://www.youtube.com/watch?v=fBGWtVOKTkM

**Pull request closed without merging**